Source/CarthageKit/Xcode.swift

```diff
@@ -947,15 +958,24 @@ public func buildInDirectory( // swiftlint:disable:this function_body_length
 						)
 						.flatMapError { _ in .empty }
 				}
-
-				return createVersionFile(
-					for: dependency.dependency,
-					version: dependency.version,
-					platforms: options.platforms,
-					buildProducts: urls,
-					rootDirectoryURL: rootDirectoryURL
-					)
-					.flatMapError { _ in .empty }
+				var copyToSharedCache = SignalProducer<(URL), CarthageError>.empty
+				if options.cacheBuilds {
+					let sharedCacheURLForDependency = sharedCacheURLByXcodeVersionFor(dependency: dependency.dependency, pinnedVersion: dependency.version)
+					copyToSharedCache = copyFrameworksAndCreateVersionFile(urls, from: rootDirectoryURL, to: sharedCacheURLForDependency,
+																		   projectName: dependency.dependency.name, pinnedVersion: dependency.version, toolchain: options.toolchain)
+							.flatMapError { _ in .empty }
+				}
+				return copyToSharedCache
+					.flatMap(.merge) { _ -> SignalProducer<(), CarthageError> in
```

Comment:
> `var copyToSharedCache = SignalProducer<(URL), CarthageError>.empty`

and

> `.flatMapError { _ in .empty }`

flatMapping to `.empty` SignalProducer will not work because there are no values to map over. `then` operator should be appropriate here.

Reply:
I'm not so comfortable with ReactiveSwift, replacing `.flatMapError { _ in .empty }` by `.then(SignalProducer<(URL), CarthageError>.empty)` fails when running `make test` :

```
$ make test
[...]
Test Case '-[CarthageKitTests.ProjectSpec buildCheckedOutDependenciesWithOptions, createAndCheckVersionFiles, should rebuild cached frameworks (and dependencies) whose version does not match the version file]' started.
2019-04-12 12:07:20.497 xctest[57360:18611164] Building scheme "TestFramework3_Mac" in TestFramework3.xcodeproj
2019-04-12 12:07:31.388 xctest[57360:18611225] Building scheme "TestFramework2_Mac" in TestFramework2.xcodeproj
2019-04-12 12:07:42.378 xctest[57360:18611164] Building scheme "TestFramework1_Mac" in TestFramework1.xcodeproj
Fatal error: 'try!' expression unexpectedly raised an error: Error Domain=NSCocoaErrorDomain Code=260 "The file “.TestFramework3.version” couldn’t be opened because there is no such file." UserInfo={NSFilePath=.build/x86_64-apple-macosx10.10/debug/CarthagePackageTests.xctest/Contents/Resources/DependencyTest/Carthage/Build/.TestFramework3.version, NSUnderlyingError=0x7f83abc83440 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}: file /BuildRoot/Library/Caches/com.apple.xbs/Sources/swiftlang_Fall2018/swiftlang_Fall2018-1000.11.42/src/swift/stdlib/public/core/ErrorType.swift, line 184
Exited with signal code 4
make: *** [test] Error 1
```

Reply:
No, I'm not saying about replacing `.flatMapError { _ in .empty }` with `.then(SignalProducer<(URL), CarthageError>.empty)`.

```swift
return copyToSharedCache.then(
  createVersionFile(
  	for: dependency.dependency,
  	version: dependency.version,
  	platforms: options.platforms,
  	buildProducts: urls,
  	rootDirectoryURL: rootDirectoryURL
  )
  .flatMapError { _ in .empty }
)
```

is what I mean.

Reply:
oh, i misread "flatMapping", sorry, I updated to change this simpler way, simpler, thanks for your review